src/group_theory/quaternion_group.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2021 Julian Kuelshammer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Kuelshammer
+-/
+import data.zmod.basic
+import group_theory.order_of_element
+import data.nat.basic
+import tactic.interval_cases
+import group_theory.dihedral
+
+/-!
+# Quaternion Groups
+
+We define the (generalised) quaternion groups `quaternion_group n` of order `4n`, also known as
+dicyclic groups, with elements `a i` and `xa i` for `i : zmod n`. The (generalised) quaternion
+groups can be defined by the presentation `⟨a, x | a²ⁿ = 1, x² = aⁿ, x⁻¹ax=a⁻¹⟩`. We write `a i` for
```

Comment:
Thinking mostly about docs here, should `⟨a, x | a²ⁿ = 1, x² = aⁿ, x⁻¹ax=a⁻¹⟩` be unicode or Latex?

Reply:
I changed to LaTeX now.